### DISCUSSION TITLE
Issue with nbconvert

### DISCUSSION THREAD

**Discussion by RWillisConger** [ucbds-infra/otter-grader#859] [Q&A] [answered]
A handful of students have issues using nbconvert to submit their assignments. They have not had issues with previous homeworks, and updating the package has not resolved the error. I can include more information or the entire file if useful, but here is the text of the error:

<img width="658" alt="Screenshot 2024-10-14 at 10 37 07 AM" src="https://github.com/user-attachments/assets/3c38a4da-02ae-486c-9fb5-4fccdc0d06ef">
<img width="457" alt="Screenshot 2024-10-14 at 10 37 38 AM" src="https://github.com/user-attachments/assets/cb830f87-1857-4e97-8cda-19918a8f4d50">

<img width="459" alt="Screenshot 2024-10-14 at 10 37 54 AM" src="https://github.com/user-attachments/assets/8bf3aa7f-0005-4389-b823-6adf07fcb750">

**Comment by chrispyles** [comment]
The LateFailed error usually happens when the LaTeX exporter generates no contents. Please send a copy of the notebook. If you don't want it posted publicly, feel free to PM me on Slack (link in README).

**Reply by RWillisConger** [reply]
Thank you! Just sent

**Reply by chrispyles** [reply, marked as answer]
Resolved on Slack.

For anyone who finds this thread, the issue was that none of the questions were marked as manually-graded (`manual: true` in question config), so when filtering the notebook for PDF export (which is the default behavior), there were no cells left to include in the PDF.

Solution is either to a) mark relevant cells with `manual: true`, b) set `export_cell: filtering: false` in the assignment config, or c) if the notebook has already been sent to students, if this is a notebook you’ve already sent out to students, you can tell them to add a new cell at the bottom with

```python
grader.export(run_tests=True, filtering=False)
```

and run that instead of the one already in the notebook. (Otter makes the cells it generates non-editable so it’s probably easier to tell students to add a new cell rather than having them edit the cell metadata.)
